File: src/RNetPi.Core/Packets/PacketS2CSourceDeleted.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x07
/// Source Deleted
/// Informs client of a deleted source
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
/// </summary>
public class PacketS2CSourceDeleted : PacketS2C
{
    public const byte ID = 0x07;

    public PacketS2CSourceDeleted(byte controllerID, byte zoneID)
    {
        Writer.Write(controllerID);
        Writer.Write(zoneID);
    }

    public override byte GetID() => ID;
}

File: src_js/server/packets/PacketS2CSourceDeleted.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x07
 * Source Deleted
 * Informs client of a deleted source
 * Data:
 *      (Unsigned Char) Source ID
 */
class PacketS2CSourceDeleted extends PacketS2C {
    constructor(ctrlrID, zoneID) {
        super();

        this._buffer.writeUInt8(ctrlrID);
        this._buffer.writeUInt8(zoneID);
    }

    getID() {
        return 0x07;
    }
}

module.exports = PacketS2CSourceDeleted;
